docs/upcoming_changes/6647.change.rst

```diff
@@ -0,0 +1,6 @@
+Support for ``@intrinsic(prefer_literal=True)``
+===============================================
+
+This PR adds ``prefer_literal`` option for the ``intrinsic`` decorator to 
+prioritize the use of literal types when available. This has the same 
+behavior as in the ``prefer_literal`` option in the ``overload`` decorator.
```

Comment:
```suggestion
In the high level extension API, the ``prefer_literal`` option is added to the
``numba.extending.intrinsic`` decorator to prioritize the use of literal types
when available. This has the same behavior as in the ``prefer_literal``
option in the ``numba.extending.overload`` decorator.
```
suggest this change, was hoping to clarify that this is an update to the high level extension API. What do you think?